habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -738,17 +758,33 @@ def get_navigable_receptacles(
             obj_mgr = sim.get_rigid_object_manager()
         receptacle_obj = obj_mgr.get_object_by_handle(receptacle.parent_object_handle)
         receptacle_bb = get_aabb(receptacle_obj.object_id, sim, transformed=True)
+
+        if receptacle_bb.size_y() > sim.pathfinder.nav_mesh_settings.agent_height - 0.2:
+            print(f"Receptacle {receptacle.parent_object_handle}, {receptacle_obj.translation} is too tall. Skipping.")
+            continue
+
+        bounds = receptacle.bounds
+        if bounds.size_x() < 0.3 or bounds.size_z() < 0.3:
+            print(f"Receptacle {receptacle.parent_object_handle}, {receptacle_obj.translation} is too small. Skipping.")
+            continue
+
+        # check if all 4 corners of the receptacle are accessible
+        gt = receptacle.get_global_transform(sim)
+        global_bounds = mn.Range3D(
+            gt.transform_point(bounds.min),
+            gt.transform_point(bounds.max),
+        )
         recep_points = [
-            receptacle_bb.back_bottom_left,
-            receptacle_bb.back_bottom_right,
-            receptacle_bb.front_bottom_left,
-            receptacle_bb.front_bottom_right
+            global_bounds.back_bottom_left,
+            global_bounds.back_bottom_right,
+            global_bounds.front_bottom_left,
+            global_bounds.front_bottom_right
         ]
-        for point in recep_points:
-            if is_accessible(sim, point, nav_to_min_distance):
-                navigable_receptacles.append(receptacle)
-                logger.info(f"Receptacle {receptacle.parent_object_handle}, {receptacle_obj.translation} is navigable.")
-                break
+        is_navigable = all(is_accessible(sim, point, nav_to_min_distance) for point in recep_points)
+
+        if is_navigable:
```

Comment:
Change from `is_navigable` to something like `all_is_accessible` since these points are not navigable, they are only accessible.